packages/frontend/src/lib/pnl.ts

```diff
@@ -0,0 +1,95 @@
+import BigNumber from 'bignumber.js'
+
+import { TransactionType } from '@constants/enums'
+import { PositionType } from '../types'
+
+interface Deposit {
+  squeethAmount: BigNumber
+  ethAmount: BigNumber
+  usdValue: BigNumber
+  timestamp: string
+  transactionType: TransactionType
+  txId: string
+  ethPriceAtDeposit: BigNumber
+}
+
+interface LongDeposit extends Deposit {
+  sellQuote: BigNumber
+}
+
+interface GetCurrentLongParams {
+  positionType: string
+  squeethAmount: BigNumber
+  transactions: any[]
+  getSellQuote: (input: BigNumber) => Promise<Record<string, any>>
+}
+
+export async function getCurrentLong({
+  getSellQuote,
+  positionType,
+  squeethAmount,
+  transactions,
+}: GetCurrentLongParams) {
+  if (positionType === PositionType.SHORT) return []
+  let totalLongSqth = new BigNumber(0)
+
+  const result: LongDeposit[] = []
+  let sellQuote
+  for (let index = 0; index < transactions.length; index++) {
+    if (totalLongSqth.gte(squeethAmount)) break
+    if (totalLongSqth.isLessThan(squeethAmount) && transactions[index].transactionType === TransactionType.BUY) {
+      totalLongSqth = totalLongSqth.plus(transactions[index].squeethAmount)
+      sellQuote = await getSellQuote(transactions[index].squeethAmount)
+      result.push({ ...transactions[index], sellQuote: sellQuote.amountOut })
+    } else if (
+      totalLongSqth.isLessThan(squeethAmount) &&
+      transactions[index].transactionType === TransactionType.SELL
+    ) {
+      totalLongSqth = totalLongSqth.minus(transactions[index].squeethAmount)
+    }
```

Comment:
What if the totalLongsqth is < squeethAmount because user transfer osqueeth to somewhere else instead of doing BUY or SELL?

Reply:
Like adding liquidity & removing liquidity and transfer osqueeth directly to other addresses